providers/ldap/ldap_client.go

```diff
@@ -641,6 +651,51 @@ func (l *LClient) TestLogin(testAuthConfig *model.TestAuthConfig) (int, error) {
 		return status, fmt.Errorf("Error in ldap bind: %v", err)
 	}
 
+	samName := username
+	if strings.Contains(username, "\\") {
+		samName = strings.SplitN(username, "\\\\", 2)[1]
+	}
+	query := "(" + testAuthConfig.AuthConfig.LdapConfig.UserLoginField + "=" + samName + ")"
+	log.Debugf("LDAP Search query: {%s}", query)
+
+	testUserSearchAttributes := []string{l.ConstantsConfig.MemberOfAttribute, l.ConstantsConfig.ObjectClassAttribute,
+		testAuthConfig.AuthConfig.LdapConfig.UserObjectClass, testAuthConfig.AuthConfig.LdapConfig.UserLoginField,
+		testAuthConfig.AuthConfig.LdapConfig.UserNameField, testAuthConfig.AuthConfig.LdapConfig.UserSearchField,
+		testAuthConfig.AuthConfig.LdapConfig.UserEnabledAttribute}
+
+	search := ldap.NewSearchRequest(testAuthConfig.AuthConfig.LdapConfig.Domain,
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+		query, testUserSearchAttributes, nil)
+
+	result, err := lConn.Search(search)
+	if err != nil {
+		return status, fmt.Errorf("Error searching the user information with new server settings: %v", err)
+	}
+
+	if len(result.Entries) < 1 {
+		return status, fmt.Errorf("Cannot locate the user information with new server settings")
+	} else if len(result.Entries) > 1 {
+		return status, fmt.Errorf("Multiple users found for the username with new server settings")
+	}
+
+	entry := result.Entries[0]
+	if !l.hasPermission(entry.Attributes, &testAuthConfig.AuthConfig.LdapConfig) {
+		return status, fmt.Errorf("User is probably disabled in the new server settings")
+	}
+
+	userIdentity, err := l.attributesToIdentity(entry.Attributes, entry.DN, l.ConstantsConfig.UserScope)
```

Comment:
Reason to call attributesToIdentity  is to make sure  changes to schema like UserLoginField/UserObjectClass also get tested
We allow admins to edit any of the schema fields during "edit settings", any changes there should also get validated
One way is to check if we are able to read all attributes as per the schema entered